homeassistant/components/august/entity.py

```diff
@@ -0,0 +1,57 @@
+"""Base class for August entity."""
+
+import logging
+
+from . import DEFAULT_NAME, DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+
+class AugustEntityMixin:
+    """Base implementation for August device."""
+
+    def __init__(self, data, device):
+        """Initialize an August device."""
+        super().__init__()
+        self._data = data
+        self._device = device
+        self._undo_dispatch_subscription = None
+
+    @property
+    def should_poll(self):
+        """Return False, updates are controlled via the hub."""
+        return False
+
+    @property
+    def _device_id(self):
+        return self._device.device_id
+
+    @property
+    def _detail(self):
+        return self._data.get_device_detail(self._device.device_id)
+
+    @property
+    def device_info(self):
+        """Return the device_info of the device."""
+        return {
+            "identifiers": {(DOMAIN, self._device_id)},
+            "name": self.name,
+            "manufacturer": DEFAULT_NAME,
+            "sw_version": self._detail.firmware_version,
+            "model": self._detail.model,
+        }
+
+    async def async_added_to_hass(self):
+        """Subscribe to updates."""
+        self._data.async_subscribe_device_id(self._device_id, self._update_from_data)
+        self._data.activity_stream.async_subscribe_device_id(
+            self._device_id, self._update_from_data
+        )
+        self._update_from_data()
```

Comment:
This will write a state

Reply:
I'm doing this to calculate the initial state. Is the issue that it is not safe or that we end up updating twice?

Reply:
we end up updating twice.

Reply:
Fixed